lib/chef/provider/package/aix.rb

```diff
@@ -87,19 +83,18 @@ def load_current_resource
 
         def candidate_version
           return @candidate_version if @candidate_version
-          status = popen4("installp -L -d #{@new_resource.source}") do |pid, stdin, stdout, stderr|
-            stdout.each_line do |line|
-              case line
-              when /\w:#{Regexp.escape(@new_resource.package_name)}:(.*)/
-                fields = line.split(":")
-                @candidate_version = fields[2]
-                @new_resource.version(fields[2])
-                Chef::Log.debug("#{@new_resource} setting install candidate version to #{@candidate_version}")
-              end
+          ret = shell_out("installp -L -d #{@new_resource.source}")
+          ret.stdout.each_line do | line |
+            case line
+            when /\w:#{Regexp.escape(@new_resource.package_name)}:(.*)/
+              fields = line.split(":")
+              @candidate_version = fields[2]
+              @new_resource.version(fields[2])
+              Chef::Log.debug("#{@new_resource} setting install candidate version to #{@candidate_version}")
             end
           end
-          unless status.exitstatus == 0
-            raise Chef::Exceptions::Package, "installp -L -d #{@new_resource.source} - #{status.inspect}!"
+          unless ret.exitstatus == 0
+            raise Chef::Exceptions::Package, "installp -L -d #{@new_resource.source} - #{ret.inspect}!"
```

Comment:
Seems like we want `ret.command` and `ret.stderr` (and maybe `ret.stdout` for good measure)